package/build_package.sh

```diff
@@ -13,29 +14,44 @@ set -euf -o pipefail
 
 # Get the package version from Stack, eliminate the single quotes.
 # Exported, because it is used with envsubst to write the control file.
-export VERSION=$(stack query locals vaultenv-real version | sed -e "s/^'//" -e "s/'$//")
+VERSION=$(stack query locals vaultenv-real version | sed -e "s/^'//" -e "s/'$//")
+export VERSION
 
 # The name of the .deb file to create
 PKGNAME="vaultenv-${VERSION}"
 
-# Recreate the file system layout as it should be on the target machine.
-mkdir -p "$PKGNAME/DEBIAN"
-mkdir -p "$PKGNAME/usr/bin"
-mkdir -p "$PKGNAME/etc/secrets.d"
-
 cd ..
 VAULTENV_NIX_PATH=$($(nix-build --no-link -A full-build-script nix/vaultenv-static.nix) | tail -n1)
 cd -
 
-cp "${VAULTENV_NIX_PATH}/bin/vaultenv" "$PKGNAME/usr/bin/"
-cp "${VAULTENV_NIX_PATH}/bin/vaultenv" "vaultenv-${VERSION}-linux-musl"
+cp --no-preserve=mode,ownership "${VAULTENV_NIX_PATH}/bin/vaultenv" "vaultenv-${VERSION}-linux-musl"
 
-# Write the package metadata file, substituting environment variables in the
-# template file.
-cat deb_control | envsubst > "$PKGNAME/DEBIAN/control"
 
 # Build the Debian package
-dpkg-deb --build "$PKGNAME"
+# We need to use `fakeroot` here to have the package files be correctly owned
+# by `root` without being `root` ourselves.
+fakeroot -- bash <<-EOFAKEROOT
```

Comment:
Why not wrap the entire `build_package.sh` in `fakeroot`?

Reply:
I didn't do this because `fakeroot`'s documentation discourages running builds under `fakeroot`:

> Also, note that it's best not to do the building of the binaries themselves under
> fakeroot. Especially configure and friends don't like it when the system suddenly
> behaves differently from what they expect. (or, they randomly unset some
> environment  variables, some of which fakeroot needs).

Reply:
Ah, I see that `build_package.sh` here also triggers Nix to build the binary, if it wasn't built before. I thought we triggered the build elsewhere and just copied the binary in this script.